5.5/mysql/bin/docker-entrypoint.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -e
+
+# TODO: Only allow mysqld_safe?
+if [ ! -d '/var/lib/mysql/mysql' -a "${1%_safe}" = 'mysqld' ]; then
+	if [ -z "$MYSQL_ROOT_PASSWORD" ]; then
+		echo >&2 'error: database is uninitialized and MYSQL_ROOT_PASSWORD not set'
+		echo >&2 '  Did you forget to add -e MYSQL_ROOT_PASSWORD=... ?'
+		exit 1
+	fi
+	
+	scl enable mysql55 -- mysql_install_db --user=mysql --datadir=/var/lib/mysql
+	
+	# These statements _must_ be on individual lines, and _must_ end with
+	# semicolons (no line breaks or comments are permitted).
+	# TODO proper SQL escaping on ALL the things D:
+	TEMP_FILE='/tmp/mysql-first-time.sql'
+	cat > "$TEMP_FILE" <<-EOSQL
+		DELETE FROM mysql.user ;
+		CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
+		GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
+		DROP DATABASE IF EXISTS test ;
+	EOSQL
+	
+	if [ "$MYSQL_DATABASE" ]; then
```

Comment:
yeah this should probably be mandatory along w/ the root and normal passwords and the user name.